### PR TITLE
Move onnx export to learners & minor train/vote/test set fixes

### PR DESCRIPTION
**Pull request by patbaa**
1.)
Onnx export requires 
 - `initial_types` for sklearn and xgboost, which defines the input data shapes
 - the knowledge if the underlying model is sklearn, xgboost, tf/keras or pytorch
 - input datasize for pytorch models

Due to these requirements, it makes more sense to let the learners define the export, whom are in possession of these information, while the wrapper in the mli_interface is not.

I've seen a JIRA issue for depricate mli*weights, for that: the onnx --> {sklearn; xgboost; pytorch; tf} conversion is needed in the future?

2.)
As there are many examples, when the vote set was introduced, a few were inconsistent. Fixes for them.

### CODE REVIEW

**Comment by patbaa** [comment]
onnx might not be suitable for us, this I close this PR. Also I make a separate PR for the dataset related minor fixes
